devito/types/basic.py

```diff
@@ -1078,11 +1078,9 @@ def indexify(self, indices=None, lshift=False, subs=None):
         subs = subs or {}
         subs = [{**{d.spacing: 1, -d.spacing: -1}, **subs} for d in self.dimensions]
 
-        # Add halo shift
-        shift = self._size_nodomain.left if lshift else tuple([0]*len(self.dimensions))
```

Comment:
uh? 😮

Reply:
Nested lower_expr takes care of it through the indexed part of it

Reply:
awesome!

Reply:
doesn't it apply to `subs` as well?

Reply:
No subs is needed because of sudomain dimension map above, because `f.indexe` don't have a subdomain